addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -226,7 +226,8 @@ export class WysiwygAdapterComponent extends ComponentAdapter {
             this.props.quitCallback({ onLeave, reloadIframe });
         };
 
-        if (!forceLeave && this._isDirty()) {
+        const isDirty = this._isDirty() && this.$editable[0].ownerDocument.defaultView.$('.o_dirty').length;
```

Comment:
This is not correct IMO. If the `_isDirty` method returns `true` when we expect it to return `false` then it is that method that should be fixed, not just adding another condition on top of it.